### PR TITLE
Adding support for scheduled campaigns on the Congrats page

### PR DESCRIPTION
**Pull request by daynew**
As a Korean student participating in the June/July Coding Party event, I want to see a link to the custom Minecraft Hero and AI Oceans certificates created by my government.

As a Platform Engineer, I want to be able to schedule custom certificates to exist for only a specific amount of time.

* Updates the extra links feature of our Congrats page to support date ranges.
* Adds the query string parameter `currentDate` so we can preview what the Congrats page will look like on a particular date.
* Adds Korean Coding Party campaign certificates for the time specified in the JIRA.

## Links
[Jira](https://codedotorg.atlassian.net/browse/P20-948)

## Video/Screenshots
https://github.com/code-dot-org/code-dot-org/assets/1372238/052740e8-7d58-4a1b-9e2c-f4ca633797be

Storybook
Screenshot 2024-06-05 at 3 45 12 PM

## Testing story
Storybook

Manual testing:
* Visit http://localhost-studio.code.org:3000/s/hero/lessons/1/levels/1/lang/ko
* Click the "I'm finished" link to view the Congrats page.
* Observe there is no "extra link" under the certificate preview.
* Append `&currentDate=2024-07-20` to the URL and visit it.
* Observe there is now a link to a custom certificate under the certificate preview image.

## Follow-up work
* Create a Jira task to delete this campaign in August
* Setup a PR to support the custom certificate for Music Lab.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/certificates/Congrats.jsx [hunk 1]
```diff
@@ -67,6 +116,15 @@ export default function Congrats(props) {
...
 
+  // Determine what time we should consider the current time to be when
+  // rendering the page. This is useful for testing UI changes which happen
+  // at particular times of the year.
+  // the currentDate can be overridden using the query string param 'currentDate'
+  const currentDateOverride =
+    hasQueryParam('currentDate') && new Date(queryParams('currentDate'));
+  // default to the real current time.
+  const currentDate = currentDateOverride || props['currentDate'] || new Date();
+
```
Comment (mgc1194) [inline apps/src/templates/certificates/Congrats.jsx:127]: I like having the ability to override the current date.

File: apps/src/templates/certificates/Congrats.jsx [hunk 2]
```diff
@@ -374,7 +432,7 @@ export default function Congrats(props) {
               isPlCourse={isPlCourse}
               userType={userType}
             >
-              {renderExtraCertificateLinks(language, tutorial)}
+              {renderExtraCertificateLinks(language, tutorial, currentDate)}
```
Comment (mgc1194) [inline apps/src/templates/certificates/Congrats.jsx:435]: The `currentDate` prop may be causing tests to fail.

```
all to console.error from "EnglishSignedOutHOCTutorial": Warning: Failed prop type: Congrats: prop type `currentDate` is invalid; it must be a function, usually from the `prop-types` package, but received `undefined`.
```

File: apps/src/templates/certificates/Congrats.jsx [hunk 3]
```diff
@@ -49,6 +55,49 @@ export default function Congrats(props) {
     );
   };
 
+  /**
```
Comment (mgc1194) [inline apps/src/templates/certificates/Congrats.jsx:58]: This is not a blocking comment, but have you considered moving the Extra certificates to a separate component?
Iw would make testing easier, and the code more readable.

**Review by mgc1194 — APPROVED** [review]
THe changes look great.
I really appreciate the documentation in the code.
I left only one question to consider.
Some unit and UI tests are complaining about the new `currentDate` prop, so you may need to update the existing test before, to get the drone build passing.

File: apps/src/templates/certificates/Congrats.jsx (hunk 1 above)
Comment (daynew) [reply apps/src/templates/certificates/Congrats.jsx:127]: Yeah this should probably be a pattern we should start following. Maybe this should actually be in redux so all components can easily adopt it. Something to think about. I won't do it in this PR.

File: apps/src/templates/certificates/Congrats.jsx (hunk 3 above)
Comment (daynew) [reply apps/src/templates/certificates/Congrats.jsx:58]: Yeah I have thought about it. Unfortunately I don't have time to do that refactoring right now :(

**Pull request merged**